_posts/2015-12-15-how-bot-named-dolores-landingham-transformed-18fs-onboarding.md

```diff
@@ -0,0 +1,281 @@
+
+3. Developing a Slack bot would provide opportunities down the line for interesting experiments, such as allowing employees to reply to the bot with questions (and receive answers!).
+
+4. We wanted a way to automate and then trickle out messages so that new hires were not overwhelmed during their first days and Slack direct messages (DMs) seemed like they would work well.
+
+For the MVP, we decided that the simplest solution possible was to set up a small Rails app with a database for storing new employee Slack names, start dates, and scheduled messages. We decided to use Rails instead of a lighter framework because we also planned to create a frontend web interface for our 18F colleagues to add and edit information without needing to understand Ruby or Git.
+
+Each scheduled message would have a “day” associated with it. One message would be for “day 1,” one for “day 2” and so on. Once per day, the app would run a job that would match employees and messages based on the “day” of the message and the start date of the employee and the Slack bot.
+
+## Naming the bot
+
+Naming is a fun and important step for any project. 18F already had a Slackbot named “slackbot”, so that name was out. We started bouncing around names, from “Onb04rd1ngb0t3000” to “Siri” to “Joanne” — but nothing felt right.
+
+Then, Greg Boone suggested “Landingham”, which was immediately greeted with a heart-eyes emoji reaction. There are many [West Wing](https://en.wikipedia.org/wiki/The_West_Wing) fans at 18F, and something felt right about bringing back the beloved character Dolores Landingham, who served as fictional President Bartlett’s wise and sometimes ornery secretary, to guide new employees through the turbulent waters of onboarding.
+
+![Screenshot of Dolores Landingham being named.](/assets/blog/onboarding/landingham-slack.jpg)
+
+Voila! `Mrs. Landingham` was born.
+
+## Building the bot
+
+We’ll spend the next section talking about bot development. If you’d like to skip that and go directly to our future plans, click here.
+
+In the [Ruby section on the Slack integrations page](https://api.slack.com/community#ruby), there are dozens of examples of Slack integrations, though none specifically addressed drip campaigns. (A drip campaign is when messages are sent to users based on a schedule or trigger event, such as number of days after being added to a Slack bot.) This meant we would have to build something from scratch rather than clone and edit an existing bot.
+
+To handle our Slack API integration, we chose the [Slack Ruby Client gem](https://github.com/dblock/slack-ruby-client), a tool for handling Slack’s web and real time messaging APIs. To start, we [added a new bot user](https://api.slack.com/bot-users) to the 18F Slack organization and configured them gem using the API token for the bot user.
+
+```ruby
+ # Gemfile
+
+ gem 'slack-ruby-client'
+
+ group :test, :development do gem 'dotenv'end
+```
+
+```ruby
+# .env
```

Comment:
can we do a new line after this? just for consistent, so each ruby file has a comment with the file name and then a new line and then the code